android/src/main/java/io/ably/lib/rest/Push.java

```diff
@@ -0,0 +1,247 @@
+                @Override
+                public void onSuccess(JsonObject response) {
+                    device.setAndPersistUpdateToken(context, response.getAsJsonPrimitive("updateToken").getAsString());
+                    rest.device = device;
+                    new ActivateCallback(context, "PUSH_ACTIVATED").onSuccess(null);
+                }
+                @Override
+                public void onError(ErrorInfo reason) {
+                    new ActivateCallback(context, "PUSH_ACTIVATED").onError(reason);
+                }
+            });
+        } else {
+            rest.device = device;
+            Http.RequestBody body = rest.http.requestBodyFromGson(device.pushMetadataJsonObject());
+            rest.asyncHttp.put("/push/deviceRegistrations/" + device.id, HttpUtils.defaultAcceptHeaders(rest.options.useBinaryProtocol), null, body, null, new ActivateCallback(context, "PUSH_RENEWED_TOKEN"));
+        }
+    }
+
+    private class ActivateCallback implements Callback<Void> {
+        private Context context;
+        private String action;
+
+        ActivateCallback(Context context, String action) {
+            this.context = context;
+            this.action = action;
+        }
+
+        @Override
+        public void onSuccess(Void v) {
+            SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
+            prefs.edit()
+                    .putBoolean(SharedPrefKeys.ACTIVATING, false)
+                    .putBoolean(SharedPrefKeys.ACTIVATED, true)
+                    .apply();
+            Intent intent = new Intent();
+            intent.setAction("io.ably.broadcast." + action);
+            intent.putExtra("failed", false);
+            LocalBroadcastManager.getInstance(context).sendBroadcast(intent);
+        }
+        @Override
+        public void onError(ErrorInfo reason) {
+            SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
+            prefs.edit()
+                    .putBoolean(SharedPrefKeys.ACTIVATING, false)
+                    .putBoolean(SharedPrefKeys.ACTIVATED, false)
+                    .apply();
+            Intent intent = new Intent();
+            intent.setAction("io.ably.broadcast." + action);
+            intent.putExtra("failed", true);
+            intent.putExtra("errorReason.message", reason.message);
+            intent.putExtra("errorReason.code", reason.code);
+            intent.putExtra("errorReason.statusCode", reason.statusCode);
+            LocalBroadcastManager.getInstance(context).sendBroadcast(intent);
+        }
+    }
+
+    public static class RegistrationToken {
+        public RegistrationTokenType type;
+        public String token;
+
+        private RegistrationToken(RegistrationTokenType type, String token) {
+            this.type = type;
+            this.token = token;
+        }
+
+        public static RegistrationToken persisted(Context context) {
```

Comment:
I am not quite sure I understand why this method is named `persisted`.  Mind explaining?